oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServletOAuth2AuthorizedClientExchangeFilterFunctionTests.java

```diff
@@ -635,6 +642,58 @@ public void filterWhenRequestAttributesNotSetAndHooksInitHooksResetThenDefaultsN
 		assertThat(getBody(request)).isEmpty();
 	}
 
+	// gh-7228
+	@Test
+	public void publisherWhenHooksInitAndSubscribedOutsideOfWebContextThenShouldNotThrowNpe() throws Exception {
+		Mono<Context> publisher = Mono.subscriberContext();
+		this.function.afterPropertiesSet();			// Hooks.onLastOperator() initialized
+//		this.function.destroy();					// Hooks.onLastOperator() released
```

Comment:
Is there a reason that the tests do not invoke .destroy()?

Reply:
leave it here just to mark that Hook is not released.  
but maybe it is better to move `this.function.destroy();` to the `cleanup()`

Reply:
moved `this.function.destroy()` to the `cleanup()` method